### DISCUSSION TITLE
Understanding Remark and Rehype Plugin Integrations

### DISCUSSION THREAD

**Discussion by enesbala5** [pngwn/MDsveX#664] [Q&A]
Hello,

I've been working with MdsveX the past few days and have been trying to use plugins from remark in order to implement features like wiki-link and TOC.

However, 90% of the time, the plugins simply did nothing. I now understand that there have been breaking changes with v13 of remark/unified, and that causes issues with compatibility.

However, my confusion grew even further once I landed here and saw that the MDsveX is moving away from the ecosystem entirely.

Can anyone please guide me on how to integrate remark packages, or something else which provides the functionality I need - because I can't seem to understand how things are supposed to work here.

Thank you!

**Comment by vvvyyynet** [comment]
+1 I'm also struggling with integrating wiki-link functionality to MDsveX and would very much appreciate any guidance and documentation. I've also tried with [Carta](https://beartocode.github.io/carta/introduction), another actively-maintained Markdown-editor based in Svelte, but there is no documentation on wiki-links and marked-plugins either. The whole compatibility issues make it very hard to understand, where to look. Also, several plugins don't seem to be actively maintained any more...

For the Obsidian and Logseq communities it would be awesome if wiki-links were covered in the documentation:
- what plugins to use (remark-plugins like [remark-wiki-link-plus](https://github.com/datopian/remark-wiki-link-plus), [remark-wiki-link](https://github.com/landakram/remark-wiki-link), ...)
- how to integrate them into MDsveX

I'm currently searching for something similar to [Quartz](https://quartz.jzhao.xyz/), but in Svelte, but it does not seem to exist... yet?!

**Comment by Sup2point0** [comment]
I'm in this exact same situation too... I've managed to get some plugins working by dropping their versions until it does something (`^2.0.0` for `remark-footnotes`, `^3.0.0` for `remark-math`), but it doesn't work for all plugins (`remark-gfm`, `remark-github-alerts`, etc.) – probably due to other dependency requirements that MDsveX can no longer fulfil =(

**Comment by Leftium** [comment]
I think I ran into the same issue trying to use [`remark-definition-list`](https://www.npmjs.com/package/remark-definition-list)
- I think I configured MDsveX correctly, but nothing happens: https://github.com/Leftium/leftium.com/commit/09fdd3aeda1710035560571e7e2a5e16f3f54645

It seems MDsveX still uses unified, but an older version.

I just need to convert an MD file, so migrating to another library like https://github.com/markdown-it/markdown-it seems to be the way to go...
